File: app/lib/core/models/clothing_item.dart

class ClothingItem {
  final int id;
  final int? userId;
  final String imageUrl;
  final String categoryGroup;
  final String? category;
  final String? description;
  final Map<String, dynamic> extras;

  ClothingItem({
    required this.id,
    this.userId,
    required this.imageUrl,
    required this.categoryGroup,
    this.category,
    this.description,
    this.extras = const {},
  });

  factory ClothingItem.fromJson(Map<String, dynamic> json) {
    final extras = Map<String, dynamic>.from(json);
    extras.remove('id');
    extras.remove('user_id');
    extras.remove('image_url');
    extras.remove('category_group');
    extras.remove('category');
    extras.remove('description');

    return ClothingItem(
      id: (json['id'] is int) ? json['id'] as int : int.tryParse('${json['id']}') ?? 0,
      userId: json['user_id'] is int ? json['user_id'] as int : (int.tryParse('${json['user_id']}') ?? null),
      imageUrl: (json['image_url'] ?? '') as String,
      categoryGroup: (json['category_group'] ?? '').toString(),
      category: json['category']?.toString(),
      description: json['description']?.toString(),
      extras: extras,
    );
  }
}

File: backend/app/Models/ClothingItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class ClothingItem extends Model
{
    // protected $table = 'products';
    protected $fillable = [
        'user_id',
        'image_url',
        'category_group',
        'category',
        'color_group',
        'description',
        'sleeve',
        'neckline',
        'fit',
        'length',
        'closure',
        'pattern',
        'material',
        'style',
        'insulation',
        'type',
        'height',
        'toe',
        'coverage',
        'penalty',
        'static_value',
    ];


    protected $casts = [
        'user_id' => 'integer',
        'image_url' => 'string',
        'category_group' => 'string',
        'category' => 'string',
        'color_group' => 'string',
        'sleeve' => 'string',
        'neckline' => 'string',
        'fit' => 'string',
        'length' => 'string',
        'closure' => 'string',
        'pattern' => 'string',
        'material' => 'string',
        'style' => 'string',
        'insulation' => 'string',
        'type' => 'string',
        'height' => 'string',
        'toe' => 'string',
        'penalty' => 'int',
        'static_value' => 'int',
    ];








    public function outfititems()
    {
        return $this->hasMany(OutfitItem::class);
    }


    public function users()
    {
        return $this->belongsTo(User::class);
    }


}
